crates/ibc/src/mock/host_helper.rs

```diff
@@ -0,0 +1,67 @@
+use alloc::format;
+use alloc::string::ToString;
+use ibc_proto::google::protobuf::Any;
+
+use crate::core::ics02_client::client_state::ClientState;
+use crate::core::ics03_connection::error::ConnectionError;
+use crate::core::ics24_host::identifier::ChainId;
+use crate::mock::client_state::MockClientState;
+use crate::Height;
+
+/// Provides an implementation of `ValidationContext::validate_self_client` for
+/// the mock host.
+pub trait ValidateSelfMockClientContext {
```

Comment:
I don't think we need this abstraction, as the `MockContext` is the only "host" that will ever need it. We can simply implement it directly in `MockContext`.

Reply:
Right [8782c93](https://github.com/cosmos/ibc-rs/pull/437/commits/8782c93a6956521334266bb045ec7f4ef09506f3)